atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+        '''
+        Extracts attribute from group object based on path, and returns default if not defined.
+        '''
+        if path:
+            section_segments = path.split('.')
+            for section in section_segments:
+                try:
+                    value = group.get(section)
+                    group = value
+                except Exception:
+                    return
+        value = group.attrs.get(attribute)
+        value = self.decode_bytes(value) if value is not None else default
+
+        return value if value is not None else default
+
+    def get_value(self, group, path: str, default=None):
+        '''
+        Extracts group or dataset from group object based on path, and returns default if not defined.
+        '''
+        section_segments = path.split('.')
+        for section in section_segments:
+            try:
+                value = group.get(section)
+                unit = self.get_attribute(group, 'unit', path=section)
+                unit_factor = self.get_attribute(group, 'unit_factor', path=section, default=1.0)
+                group = value
+            except Exception:
+                return
+
+        if value is None:
+            value = default
+        elif isinstance(value, h5py.Dataset):
+            value = value[()]
+            value = self.apply_unit(value, unit, unit_factor)
+        value = self.decode_bytes(value)
+
+        return value if value is not None else default
+
+    def parse(self, path: str = None, **kwargs):
+        source = kwargs.get('source', self.filehdf5)
+        isattr = kwargs.get('isattr', False)
+        value = None
+        if isattr:
+            attr_path, attribute = path.rsplit('.', 1)
+            value = self.get_attribute(source, attribute, path=attr_path)
+        else:
+            value = self.get_value(source, path)
+        self._results[path] = value
+
+class H5MDParser(MDParser):
+    def __init__(self):
+        super().__init__()
+        self._data_parser = HDF5Parser()
+        self._n_frames = None
+        self._n_atoms = None
+        self._atom_parameters = None
+        self._system_info = None
+        self._observable_info = None
+        self._parameter_info = None
+        self._time_unit = ureg.picosecond
+        self._path_group_particles_all = None
+        self._path_group_positions_all = None
+        self._path_value_positions_all = None
+
+        self._nomad_to_particles_group_map = {
+            'positions': 'position',
+            'velocities': 'velocity',
+            'forces': 'force',
+            'labels': 'species_label',
+            'label': 'force_field_label',
+            'mass': 'mass',
+            'charge': 'charge',
+        }
+
+        self._nomad_to_box_group_map = {
+            'lattice_vectors': 'edges',
+            'periodic': 'boundary',
+            'dimension': 'dimension'
+        }
+
+    def format_times(self, times):
```

Comment:
i do not see any references to this.

Reply:
yes, thanks, this is an artifact from something we changed. removed.